zipline/pipeline/loaders/utils.py

```diff
@@ -2,150 +2,85 @@
 
 import numpy as np
 import pandas as pd
-from six import iteritems
-from six.moves import zip
-
-from zipline.utils.numpy_utils import categorical_dtype, NaTns
 from zipline.utils.pandas_utils import mask_between_time
 
 
-def next_event_frame(events_by_sid,
-                     dates,
-                     missing_value,
-                     field_dtype,
-                     event_date_field_name,
-                     return_field_name):
+def is_sorted_ascending(a):
+    return (np.fmax.accumulate(a) <= a).all()
+
+
+def next_event_indexer(all_dates,
+                       all_sids,
+                       event_dates,
+                       event_timestamps,
+                       event_sids):
     """
-    Make a DataFrame representing the simulated next known dates or values
-    for an event.
+    Construct an indexer that will convert a column of event data into a 2D
+    array containing a value for the most recent next-known event for each sid
+    on each date.
 
-    Parameters
-    ----------
-    dates : pd.DatetimeIndex.
-        The index of the returned DataFrame.
-    events_by_sid : dict[int -> pd.Series]
-        Dict mapping sids to a series of dates. Each k:v pair of the series
-        represents the date we learned of the event mapping to the date the
-        event will occur.
-    event_date_field_name : str
-        The name of the date field that marks when the event occurred.
+    Locations where no next event was known will be filled with -1.
+    """
+    assert is_sorted_ascending(event_dates), "event dates must be sorted"
+    assert len(event_sids) == len(event_dates) == len(event_timestamps), \
```

Comment:
We should probably refactor such that we make the same assertions in `prev_event_indexer` as we do here.